src/slackapp/flows/createDeliveryRequest.js

```diff
@@ -0,0 +1,379 @@
+const { sortBy, isEqual } = require("lodash");
+const slackapi = require("../../slackapi");
+const { findChannelByName } = require("../lib/channels");
+const { errorResponse, errorView } = require("../views");
+const {
+  findOpenRequests,
+  findRequestByCode,
+  updateRequestByCode
+} = require("../../airtable");
+
+const primaryChannelName = "community_needs";
+const modalTitle = "Create Delivery Request";
+
+// Slack callback_ids -these functions are hoisted from below for readability
+selectRequestForSending.id = "select_delivery_needed_request";
+draftRequest.id = "draft_delivery_needed_message";
+draftConfirm.id = "confirm_delivery_needed_message";
+sendMessage.id = "send_delivry_neeed_message";
+
+/**
+ * Adds the interaction handlers for this flow to the slack app.
+ * This flow handles the "Post Open Request" shortcut available for
+ * intake volunteers to draft and send a new post asking for delivery volunteers
+ * from a Request.
+ *
+ * Slack Setup:
+ * - An interactive message Shortcut with callbackId: select_delivery_needed_request configured in the slack app.
+ * - Scopes: channels:history channels:write chat:write chat:write_public commands
```

Comment:
I think we should keep these scopes in one file. Right now theres a partial list in https://github.com/crownheightsaid/mutual-aid-app/blob/master/SETUP.md that I need to update